cdap-authorization-service/src/main/java/co/cask/cdap/authorization/ACLManagerHandler.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.authorization;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.common.authorization.ObjectIds;
+import co.cask.common.authorization.ACLEntry;
+import co.cask.common.authorization.ACLStore;
+import co.cask.common.authorization.ObjectId;
+import co.cask.common.authorization.Permission;
+import co.cask.common.authorization.SubjectId;
+import co.cask.common.authorization.client.ACLStoreSupplier;
+import co.cask.http.AbstractHttpHandler;
+import co.cask.http.HttpResponder;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Set;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+/**
+ * Handler for getting and setting ACL entries.
+ *
+ * TODO: SSL and authorization for this
+ */
+@Path("/v1")
+public class ACLManagerHandler extends AbstractHttpHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ACLManagerHandler.class);
+  private static final Gson GSON = new Gson();
+
+  private final Supplier<ACLStore> aclStoreSupplier;
+
+  @Inject
+  public ACLManagerHandler(ACLStoreSupplier aclStoreSupplier) {
+    this.aclStoreSupplier = aclStoreSupplier;
+  }
+
+  public ACLManagerHandler(ACLStore aclStore) {
+    this.aclStoreSupplier = Suppliers.ofInstance(aclStore);
+  }
+
+  @GET
+  @Path("/acls/global")
+  public void getGlobalACLs(HttpRequest request, HttpResponder responder,
+                            @QueryParam("subject") String subjectString,
+                            @QueryParam("permission") String permissionString) {
+
+    // TODO: validate input
```

Comment:
Looks like you are validating input. This and other TODOs below can be removed, right?